### ISSUE TITLE
Regression in async debugging with 1.22

### ISSUE THREAD

**Issue by jwren** [dart-lang/sdk#28770]
Before SDK version 1.22, breakpoint at line 5 would break, now line 4 breaks, but not line 5:

<img width="370" alt="screen shot 2017-02-14 at 4 37 09 pm" src="https://cloud.githubusercontent.com/assets/1533520/22956631/17f8035c-f2d8-11e6-9c4e-f5781d2aaac5.png">

Confirmed in SDK version 1.22.0-dev-10.5 and in SDK version 1.22 proper.

Confirmed that in version 1.21.1, line 5 is hit by debugger.

Confirmed that issue exists in IJ 2016.3.4 and in the latest bleeding edge of the Dart Plugin for IntelliJ source (as of 2-14-2017).

**Comment by devoncarew** [comment]
I can repro this In IntelliJ (w/ the latest sdk from head) and in observatory.

For convenience:

```
import 'dart:async';

main(List<String> args) async {
  Future<String> foo() async => 'hello';
  print(await foo());
}
```

**Comment by a-siva** [comment]
@rmacnak-google any ideas what changed recently to cause this

**Comment by kevmoo** [comment]
Dupe of https://github.com/dart-lang/sdk/issues/27242

**Comment by devoncarew** [comment]
@kevmoo, the other issue was about not breaking in uncaught exceptions - this is about breakpoints.

**Comment by kevmoo** [comment]
Oops! Wonder if they are related. Will reopen...

**Comment by rmacnak-google** [comment]
I can't think of any recent changes that should affect this.

This doesn't reproduce for me. Locally I see the program finish without hitting any breakpoint :(

**Comment by a-siva** [comment]
I am able to reproduce it with observatory, it breaks at line #4 but not at line #5 
If I set a breakpoint at main.dart:5:3 (i.e also specifiy the column number corresponding to 'print'
then the breakpoint at line 5 is hit) however a breakpoint at main.dart:5:9(column numner
corresponding to await) then it does not stop.

**Comment by devoncarew** [comment]
I ran into this when debugging an app last night - I had to modify the source - to re-write w/o using await - to continue debugging. I think this will affect users enough to be worth patching a fix into 1.22 stable.

**Comment by devoncarew** [comment]
@a-siva @johnmccutchan, adding this to 1.23; this seems important to have fixed.

**Comment by mhausner** [comment]
The behavior has changed, but there is still a bug.

Setting a breakpoint at line 5 by clicking on the line number in Observatory sets a breakpoint at line 5, column 9 (the await operator). Resuming the program then results in the breakpoint on line 3 col 5 being hit again. Resuming again then finishes the program without breaking at line 5.

If I set the breakpoint at line 5, column 3 using the Observatory command "b d.dart:5:3" then the debugger stops at line 3 (like above) and also at line 5.

In no case did the debugger stop at line 4 though.

**Comment by mhausner** [comment]
Fix is on the way.

**Comment by mhausner** [comment]
This is fixed in 408a64756e166b884840482194600169d39784fb.
